### PR TITLE
gRPC + REST endpoints

### PR DESCRIPTION
**Pull request by LuaxY**
## Description

Add missing REST endpoints and generate gRPC gateway endpoints

### General
- Create proto tools dependencies installer script
- Fix `GetFeedByFeedId` missing `feedId` parameter
- Add missing `RequestNewRoundTx` rpc
- Fix cmd.sh `--from` parameter due to unauthorized user
- Simplify `handler.go` (lot of repeating code)
- Add gRPC-gateway swagger file generation script for future use

### REST (Querier)
- Split into 3 files: `rest/rest.go`, `rest/query.go` and `rest/tx.go`
- Add `get module owner` and `get feed info` endpoints
- Add `/legacy` to endpoints path to avoid conflicts with gRPC-gateway
- Fix custom routing key in existing endpoints (use `types.QuerierRoute`)

### gRPC & gRPC-gateway
- Add generation of gRPC-gateway files (`*.pb.gw.go`) in `protogen`
- Register `MsgSever` in `RegisterSevices()` in `module.go`
- Register gRPC-gateway routes in `module.go`
- Avoid repeating code for `RegisterImplementations()` in `codec.go`
- Prepare `RegisterInterface()` for exported data (RoundData, Observation, OCRAbiEncoded)
- Register MsgServiceDesc
- Refactor msgServer implementation to follow [Cosmos SDK documentation](https://docs.cosmos.network/v0.42/migrations/app_and_modules.html#wiring-up-msg-and-query-services)

#### Closes: #79

### CODE REVIEW

File: scripts/protoc-swagger-gen.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
```
Comment (araskachoi) [inline scripts/protoc-swagger-gen.sh:1]: nice bash code!

File: scripts/proto-tools-installer.sh [hunk 2]
```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+
+set -ue
```
Comment (araskachoi) [inline scripts/proto-tools-installer.sh:3]: wow! very nice bash code here! 
kind of a nit, but could include a check to see if protoc is already installed, but it wouldn't matter at all if we didnt

File: x/chainlink/module.go [hunk 3]
```diff
@@ -84,7 +86,10 @@ func (AppModuleBasic) RegisterRESTRoutes(clientCtx client.Context, rtr *mux.Rout
 
 // RegisterGRPCGatewayRoutes registers the gRPC Gateway routes for the module.
 func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {
-	// TODO: enable this using protoc-gen-grpc-gateway
+	err := types.RegisterQueryHandlerClient(context.Background(), mux, types.NewQueryClient(clientCtx))
+	if err != nil {
+		panic(err) // FIXME
```
Comment (freddyli7) [inline x/chainlink/module.go:91]: `RegisterGRPCGatewayRoutes` will be called upon the module gets initiated, so I guess it's fine to panic here, what you think?

File: x/chainlink/exported/exported.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
+// Copyright 2021 ChainSafe Systems
+// SPDX-License-Identifier: MIT
+
+package exported
```
Comment (freddyli7) [inline x/chainlink/exported/exported.go:4]: just out of curiosity, what are these interfaces for?

Comment (LuaxY) [reply x/chainlink/exported/exported.go:4]: You have the ability to export some interface to allow developer of an Cosmos App to read your data easily
eg: https://github.com/cosmos/cosmos-sdk/blob/master/x/bank/exported/exported.go

File: x/chainlink/module.go (hunk 3 above)
Comment (LuaxY) [reply x/chainlink/module.go:91]: Think is fine to panic here too, and I can't see other alternative to handle the error

Comment (freddyli7) [reply x/chainlink/module.go:91]: cool, I saw you added a comment there saying `FIXME`, that's why I asked, but it is correct.

File: x/chainlink/exported/exported.go (hunk 4 above)
Comment (freddyli7) [reply x/chainlink/exported/exported.go:4]: I see, we need to export the data in our module for another modules to use for sure. I assume that those commented out interfaces registered in `RegisterInterfaces` are also needed.

**Review by freddyli7 — APPROVED** [review]
LGTM!

File: x/chainlink/module.go (hunk 3 above)
Comment (araskachoi) [reply x/chainlink/module.go:91]: yeah i think panic is fine here too

File: scripts/proto-tools-installer.sh (hunk 2 above)
Comment (araskachoi) [reply scripts/proto-tools-installer.sh:3]: maybe something like:
```
which protoc > /dev/null
if [ $? == 1  ]; then
  return 1
fi
  
```

**Review by araskachoi — APPROVED** [review]
lgtm

File: x/chainlink/exported/exported.go (hunk 4 above)
Comment (LuaxY) [reply x/chainlink/exported/exported.go:4]: Yep exact :)

**Pull request merged**